plugins/IPv4/plugin.go

```diff
@@ -0,0 +1,297 @@
+	case dhcpv4.MessageTypeRequest:
+		reply, err = dhcpv4.NewReplyFromRequest(req, dhcpv4.WithMessageType(dhcpv4.MessageTypeAck), dhcpv4.WithYourIP(ipaddr), dhcpv4.WithNetmask(ClientSubnet), dhcpv4.WithRouter(serverIP), dhcpv4.WithDNS(DNSServer), dhcpv4.WithLeaseTime(LeaseTime))
+		if err != nil {
+			log.Printf("plugins/IPv4: NewReplyFromRequest failed: %v", err)
+		}
+		reply.UpdateOption(dhcpv4.OptServerIdentifier(serverIP))
+	default:
+		log.Printf("plugins/IPv4: Unhandled message type: %v", mt)
+	}
+	return reply, true
+}
+
+// setupIPV6 not implemented only IPv4
+func setupIPV6(args ...string) (handler.Handler6, error) {
+	return nil, nil
+}
+
+func setupIPv4(args ...string) (handler.Handler4, error) {
+	_, h4, err := setupIP(false, args...)
+	return h4, err
+}
+
+func setupIP(v6 bool, args ...string) (handler.Handler6, handler.Handler4, error) {
+	if v6 {
+
+	} else {
+		if len(args) < 4 {
+			return nil, nil, errors.New("plugins/IPv4: need a file name, server IP, netmask and a DNS server")
+		}
+		filename = args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/IPv4: got empty file name")
+		}
+		serverIP = net.ParseIP(args[1])
+		if serverIP.To16() == nil {
+			return Handler6, Handler4, errors.New("plugins/IPv4: expected an IPv4 address, got: " + args[1])
+		}
+		netmask = net.ParseIP(args[2])
+		if netmask.To16() == nil {
+			return Handler6, Handler4, errors.New("plugins/IPv4: expected an IPv4 address, got: " + args[2])
+		}
+		if netmask.IsUnspecified() {
+			return Handler6, Handler4, errors.New("plugins/IPv4: netmask can not be 0.0.0.0, got: " + args[2])
+		}
+		if !checkValidNetmask(netmask) {
+			return Handler6, Handler4, errors.New("plugins/IPv4: netmask is not valid, got: " + args[2])
+		}
+		DNSServer = net.ParseIP(args[3])
+		if DNSServer.To16() == nil {
+			return Handler6, Handler4, errors.New("plugins/IPv4: expected an IPv4 address, got: " + args[3])
+		}
+		leaseTime, err := strconv.ParseUint(args[4], 10, 32)
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/IPv4: expected an uint32, got: " + args[4])
+		}
+		LeaseTime = uint32(leaseTime)
+		subnet := net.ParseIP(args[5])
+		if subnet.To16() == nil {
+			return Handler6, Handler4, errors.New("plugins/IPv4: expected an IPv4 address, got:" + ClientSubnet.String())
+		}
+		subnet = subnet.To4()
+		ClientSubnet = net.IPv4Mask(subnet[0], subnet[1], subnet[2], subnet[3])
+		records, err := LoadDHCPv4Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/IPv4: failed to load DHCPv4 records: %v", err)
+		}
+		StaticRecords = records
+		log.Printf("plugins/IPv4: loaded %d leases from %s", len(StaticRecords), filename)
+	}
+
+	return Handler6, Handler4, nil
+}
+func createIP(serverIP net.IP, netmask net.IP) (Record, bool) {
+
+	rand.Seed(time.Now().Unix())
+	ipserver := serverIP.To4()
+	mask := netmask.To4()
+	ip := net.IPv4(random(1, 254), random(1, 254), random(1, 254), random(1, 254)).To4()
```

Comment:
Is this allocating a random IP? Does this fits your needs? What is your use case?

Shouldn't there be a configuration with a subnet from where to get IPs?